### PR TITLE
Provide common information about systemd

### PR DESCRIPTION
**Pull request by matejmatuska**
Two new actors, `ScanSystemdServicesSource` for the source system and `ScanSystemServicesTarget` for the target system, provide information about services via two message types : 
- `SystemdServicesInfo(Source|Target)` - provides info about services unit files
- `SystemdServicesPresetInfoSource(Source|Target)` - provides information about vendor preset states

The actors are very similar so the shared code is in a shared library `systemd`.

Existing `systemctl` calls are not replaced by the messages, this will be done at a different time in a different PR. This PR just lays the foundation for systemd service handling.

### Examples
Example message contents (selected interesting lines):
```
[
  {
  "actor": "scan_systemd_services",
  "topic": "system_info",
  "message": {
  "hash": "15bd151fea011ea031eb130d601d3b757c8a0e1f191338fdf0988e15559a6076", 
  "data": "{\"service_files\": [
       {\"enabled_state\": \"enabled\", \"name\": \"abrt-ccpp.service\"},
       {\"enabled_state\": \"indirect\", \"name\": \"systemd-readahead-done.service\"},
       {\"enabled_state\": \"enabled\", \"name\": \"tuned.service\"},
       {\"enabled_state\": \"static\", \"name\": \"vdo-start-by-dev@.service\"},
       {\"enabled_state\": \"enabled\", \"name\": \"vdo.service\"},
       {\"enabled_state\": \"enabled\", \"name\": \"getty@.service\"},
       {\"enabled_state\": \"disabled\", \"name\": \"wpa_supplicant.service\"}
     ],
     \"services\": [
       {\"loaded\": \"loaded\", \"name\": \"abrt-ccpp.service\", \"state\": \"active\", \"unit_file\": {\"enabled_state\": \"enabled\", \"name\": \"abrt-ccpp.service\"}},
       {\"loaded\": \"loaded\", \"name\": \"systemd-readahead-done.service\", \"state\": \"inactive\", \"unit_file\": {\"enabled_state\": \"indirect\", \"name\": \"systemd-readahead-done.service\"}},
       {\"loaded\": \"loaded\", \"name\": \"vdo.service\", \"state\": \"active\", \"unit_file\": {\"enabled_state\": \"enabled\", \"name\": \"vdo.service\"}},
       {\"loaded\": \"loaded\", \"name\": \"tuned.service\", \"state\": \"active\", \"unit_file\": {\"enabled_state\": \"enabled\", \"name\": \"tuned.service\"}},
       {\"loaded\": \"not-found\", \"name\": \"ntpd.service\", \"state\": \"inactive\", \"unit_file\": null},
       {\"loaded\": \"loaded\", \"name\": \"getty@tty1.service\", \"state\": \"active\", \"unit_file\": {\"enabled_state\": \"enabled\", \"name\": \"getty@.service\"}}
     ]
   },
   "type": "SystemdServicesInfoSource"
  },
  {
    "actor": "scan_systemd_services", 
    "topic": "system_info", 
    "message": {
      "hash": "657d2987b1564798cfb2af603c22fb84aba65f65afea3161edbb7a1c63f706d3", 
      "data": "{\"presets\": [{\"service\": \"abrt-xorg.service\", \"state\": \"enable\"},
          {\"service\": \"sshd-keygen.service\", \"state\": \"disable\"},
          {\"service\": \"systemd-readahead-collect.service\", \"state\": \"enable\"},
          {\"service\": \"systemd-nspawn@.service\", \"state\": \"disable\"},
          {\"service\": \"systemd-rfkill@.service\", \"state\": \"disable\"},
          {\"service\": \"atd.service\", \"state\": \"enable\"},
          {\"service\": \"systemd-readahead-done.service\", \"state\": \"enable\"},
          {\"service\": \"getty@tty1.service\", \"state\": \"disable\"},
          {\"service\": \"firewalld.service\", \"state\": \"enable\"},
          {\"service\": \"wpa_supplicant.service\", \"state\": \"disable\"},
          ...
    },

    "type": "SystemdServicesPresetInfoSource"
  }
]
```

Jira ref.: OAMG-1812

### CODE REVIEW

File: repos/system_upgrade/common/models/systemdservices.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        'linked-runtime',
+        'alias',
+        'masked',
+        'masked-runtime',
+        'static',
+        'disabled',
+        'indirect',
+        'generated',
+        'transient',
+        'bad'
```
Comment (pirat89) [inline repos/system_upgrade/common/models/systemdservices.py:26]: sort

**Review by pirat89 — CHANGES_REQUESTED** [review]
Seems good to me, discovered mostly just minor (docstrings, ..). I need to think about couple of things yet and do some changes before the merge (+ testing & squash).

note: rebased + added a commit with todo/fixme notes and mini refactoring (I choosed this time this way instead of comment in the webpage. all notes will be dropped when done).

**Comment by pirat89** [comment]
@matejmatuska the second commit is supposed to be squashed into the first one. I've been thinking about separating it into more commits but it seems simple enough to actually have just one commit. I did the split of the error handling into the actors in the end as it seems to me more clean, adding info about the raised exceptions into docstrings.

Bigger change is around the *target* msgs. I've decided in in case of errors to not produce particular target msgs so it's easy to desntiguish between valid and invalid content - hence, this is still discussable part. So I plan to merge that tomorrow after someone else confirm it's the way to go (ping: @oamg/developers )

**Review by Rezney — APPROVED** [review]

**Pull request merged**